lib/invalidScopeDisables.js

```diff
@@ -0,0 +1,67 @@
+/* @flow */
+"use strict";
+
+/*:: type rangeDataType = {
+  all: Array<Object>,
+}
+*/
+
+/*:: type rangeType = {
+  unusedRule: string,
+  end?: number,
+  start: number,
+  used?: boolean,
+}*/
+
+/*:: type unusedRangeT = {
+  start: number,
+  end?: number,
+  strictStart: boolean,
+  strictEnd?: boolean
+}*/
+
+module.exports = function(
+  results /*: Array<stylelint$result>*/,
+  config /*: stylelint$config*/
+) /*: stylelint$disableOptionsReport*/ {
+  const report = [];
+  const usedRules = new Set(Object.keys(config.rules || {}));
+
+  usedRules.add("all");
+
+  results.forEach(result => {
+    // File with `CssSyntaxError` have not `_postcssResult`
+    if (!result._postcssResult) {
+      return;
+    }
+
+    const sourceReport = { source: result.source, ranges: [] };
+    const rangeData /*: rangeDataType*/ =
+      result._postcssResult.stylelint.disabledRanges;
+    const disabledRules = Object.keys(rangeData);
+
+    disabledRules.forEach(rule => {
+      if (usedRules.has(rule)) {
+        return;
+      }
+
+      rangeData[rule].forEach((range /*: unusedRangeT */) => {
+        if (!range.strictStart && !range.strictEnd) {
```

Comment:
if current range is not strict from start or end, then skip it, e.g:
```
/* stylelint-disable */ a { color: red} /* stylelint-enable */
/* stylelint-disable color-hex */ a { color: red} /* stylelint-enable */
```

current code will skip creating warning for `color-hex`